back-end/API/app/repositories/jobOffer_repo.py

```
@@ -1,115 +1,142 @@
+        programs = db.session.scalars(
+            db.select(StudyProgram).where(StudyProgram.id.in_(study_program_ids))
+        ).all()
+        newJobOffer.studyPrograms = programs
+
+    def _set_employment_schedules_to_model(self, newJobOffer, schedule_ids: list[int]) -> None:
+        schedules = db.session.scalars(
+            db.select(EmploymentSchedule).where(EmploymentSchedule.id.in_(schedule_ids))
+        ).all()
+        newJobOffer.employmentSchedules = schedules
+
+    def delete_by_id(self, id) -> None:
```

Ruff (0.15.13)

[error] Function argument `id` is shadowing a Python builtin

(A002)